Mastodon/Scene/Profile/Following/FollowingListViewModel+State.swift

```diff
@@ -117,58 +124,72 @@ extension FollowingListViewModel.State {
                 maxID = nil
             }
             
-            guard let viewModel = viewModel, let stateMachine = stateMachine else { return }
+            guard let viewModel, let stateMachine else { return }
             
-            guard let userID = viewModel.userID, !userID.isEmpty else {
+            guard let userID = viewModel.userID, userID.isEmpty == false else {
                 stateMachine.enter(Fail.self)
                 return
             }
             
             Task {
                 do {
-                    let response = try await viewModel.context.apiService.following(
+                    let accountResponse = try await viewModel.context.apiService.following(
                         userID: userID,
                         maxID: maxID,
                         authenticationBox: viewModel.authContext.mastodonAuthenticationBox
                     )
 
                     var hasNewAppend = false
-                    var userIDs = viewModel.userFetchedResultsController.userIDs
-                    for user in response.value {
-                        guard !userIDs.contains(user.id) else { continue }
-                        userIDs.append(user.id)
+
```

Comment:
I've run into an issue where there was a recursion in this state machine, leading to repeated calls to the relationship and followers APIs, in my case the followers list was empty so a call to the relationships was redundant, you might want to consider something like the following:

```suggestion
                    if accountResponse.value.isEmpty {
                        viewModel.accounts = []
                        viewModel.relationships = []
                        return await enter(state: Idle.self)
                    }
                    
```

Reply:
Ah, great, I'll have a look at it, thank you!

Reply:
Unfortunately, this seems to be only half of the fix, the other half seems to be running the `ListBatchFetchViewModel` running amok — for whatever reason. I'll need to wrap my head around. Good, that you fetched this annoying bug!!

Reply:
Regarding the recursion: Great suggestion with entering the `Idle`-state, I picked `NoMore` so that it shows the "Follows from another servers"-text again.

And, as I said: I wondered why those requests keep getting sent every second or so. Turns out: There's a timer in `ListBatchFetchViewModel` which fires every second, that's why the activity indicator kept spinning. To me, this looks like a hack: Every second, this timer checks, if a refresh is needed.

I'll replace it for this very screen with another approach (did someone say `UIScrollViewDelegate`?), but using this timer-based approach seems to be quite a hack to me. We should think about a viable solution for the other screens.

Reply:
Good findings! Sounds like we're getting somewhere. And yes I think we should move to a more reactive approach instead of a polling timer.